pwa-devdocs/_drafts/configuration-management/index.md

```diff
@@ -0,0 +1,431 @@
+                                            development, which enables all PWA
+                                            features.
+  buildpack create-env-file <directory>     Generate a .env file in the provided
+                                            directory to store project
+                                            configuration as environment
+                                            variables
+  buildpack load-env <directory>            Load and validate the current
+                                            environment, including .env file if
+                                            present, to ensure all required
+                                            configuration is in place.
+
+Options:
+  --version  Show version number                                       [boolean]
+  --help     Show help                                                 [boolean]
+```
+
+A project with `@magento/pwa-buildpack` installed may use the `buildpack` command in NPM scripts.
+
+```json
+"scripts": {
+  "load-env": "buildpack load-env ."
+}
+```
+
+With such a configuration, you could run `npm run-script load-env` or `yarn run load-env` to use the local copy of the Buildpack CLI.
+
+You can also invoke `buildpack` directly using the `npx` tool, which installs packages and runs their CLIs in a single command:
+
+```sh
+npx @magento/pwa-buildpack <command>
+```
+
+_Note: It is not recommended to globally install buildpack with `yarn global add`._
+_Individual projects should use their own versions and a globally installed buildpack can cause unexpected behavior._
+
+### `buildpack create-env-file <dir>`
+
+When creating a new project, or when cloning PWA Studio for the first time to contribute directly to the codebase, you must create a `.env` file in order to build Venia.
+The `create-env-file` subcommand makes that easy:
+
+```sh
+npx @magento/pwa-buildpack create-env-file .
+```
+
+### `buildpack load-env <dir>`
+
+The `load-env` subcommand loads and validates the local environment according to the [Project Environment Definitions](#project-environment-definitions), including deprecated and changed settings.
+A command, shell script, or spawned subprocess can override individual environment variables at start time.
+When loading from `.env`, Buildpack will not overwrite variables that have already been declared.
+The command does not require a `.env` file to be present; however, if one is _not_ present, _and_ the variable `NODE_ENV` is _not_ set to `production`, then it will log a warning.
+If a `.env` file is not present, the environment may still be valid, because required variables may already be set by another process or command.
+
+```sh
+npx @magento/pwa-buildpack load-env .
+```
+
+_Note: It would be unusual to use the above example directly; the more common use of `buildpack load-env` is in NPM scripts as described above._
+
+### `buildpack create-custom-origin <dir>`
```

Comment:
The `<dir>` argument is pretty straightforward for the other two commands(create in this directory or load from this directory), but what is the significance of `<div>` for this command?

Reply:
Buildpack creates the custom origin subdomain name based on a hash of the directory path. In general, we want to explicitly pass directory references to Buildpack commands because so many of them are reliant on a PWA directory for context. In this case it's especially relevant because the custom origin hash will change depending on the string path of the directory you're in.